typescript/sdk/src/ism/types.ts

```diff
@@ -122,11 +123,25 @@ export type NullIsmConfig =
   | OpStackIsmConfig
   | TrustedRelayerIsmConfig;
 
-export type RoutingIsmConfig = OwnableConfig & {
-  type: IsmType.ROUTING | IsmType.FALLBACK_ROUTING;
-  domains: ChainMap<IsmConfig>;
+type BaseRoutingIsmConfig<
+  T extends
+    | IsmType.ROUTING
+    | IsmType.FALLBACK_ROUTING
+    | IsmType.ICA_FALLBACK_ROUTING,
+> = {
+  type: T;
 };
 
+export type DomainRoutingIsmConfig = BaseRoutingIsmConfig<
+  IsmType.ROUTING | IsmType.FALLBACK_ROUTING
+> &
+  OwnableConfig & { domains: ChainMap<IsmConfig> };
+
+export type IcaRoutingIsmConfig =
+  BaseRoutingIsmConfig<IsmType.ICA_FALLBACK_ROUTING>;
```

Comment:
these should be inferred from the zod schema so we dont have multiple definitions